### PR TITLE
feat: childrenTag prop for SfListItem

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes # https://vsf.atlassian.net/browse/SFUI2-1082

# Scope of work

childrenTag prop for SfListItem 

# Screenshots of visual changes

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

**Review by AdamPawlinski — APPROVED** [review]
Good job :+1:

File: packages/sfui/frameworks/react/components/SfListItem/types.ts [hunk 1]
```diff
@@ -10,4 +10,5 @@ export interface SfListItemProps extends PropsWithChildren, PropsWithStyle {
   slotSuffix?: ReactNode;
   slotPrefix?: ReactNode;
   role?: string;
+  childrenTag?: ElementType<any> | undefined;
```
Comment (dkacper) [inline packages/sfui/frameworks/react/components/SfListItem/types.ts:13]: By default `ElementType` sets `any` as an argument, so you don't have to pass it explicitly. Also, `?` makes property optional, so there's no need to set `| undefined`.
```suggestion
  childrenTag?: ElementType;
```

File: packages/tests/components/SfListItem/SfListItem.PageObject.ts [hunk 2]
```diff
@@ -8,6 +8,13 @@ export default class SfListItemObject extends BasePage {
     return this;
   }
 
+  childrenHaveTag(tagName: string) {
+    this.container.then((el) => {
+      expect(el[0].children[0].tagName).to.equal(tagName);
```
Comment (dkacper) [inline packages/tests/components/SfListItem/SfListItem.PageObject.ts:13]: This test checks if first children's tag is as expected. According to the implementation all 3 children must match this tag. Could you iterate through children and test all of them? I know prefix and suffix are optional, but we should test them too.

[`each`](https://docs.cypress.io/api/commands/each) method may help you.

Comment (justyna-13) [reply packages/tests/components/SfListItem/SfListItem.PageObject.ts]: https://github.com/vuestorefront/storefront-ui/pull/2888/commits/63c0c62048b89164de41862f6a610a5acb1f2059

File: packages/sfui/frameworks/react/components/SfListItem/types.ts (hunk 1 above)
Comment (justyna-13) [reply packages/sfui/frameworks/react/components/SfListItem/types.ts]: https://github.com/vuestorefront/storefront-ui/pull/2888/commits/63c0c62048b89164de41862f6a610a5acb1f2059

File: packages/tests/components/SfListItem/SfListItem.PageObject.ts [hunk 3]
```diff
@@ -8,6 +8,18 @@ export default class SfListItemObject extends BasePage {
...
 
+  childrenHaveTag(tagName: string) {
+    this.container.then((el) => {
+      cy.wrap(el)
+        .children()
+        .each((child) => {
+          const childTagName = child[0].tagName;
+          expect(childTagName).to.equal(tagName);
+        });
+    });
```
Comment (FRSgit) [inline packages/tests/components/SfListItem/SfListItem.PageObject.ts:19]:
```suggestion
    this.container
      .children()
      .each((child) => {
        const childTagName = child[0].tagName;
        expect(childTagName).to.equal(tagName);
      });
```

Comment (justyna-13) [reply packages/tests/components/SfListItem/SfListItem.PageObject.ts]: https://github.com/vuestorefront/storefront-ui/pull/2888/commits/ebc0e570c7849205d67350e4b781649535f1d141

File: packages/tests/components/SfListItem/SfListItem.cy.tsx [hunk 4]
```diff
@@ -123,6 +126,16 @@ describe('SfListItem', () => {
     });
   });
 
+  describe('when childrenTag is set to div', () => {
+    before(() => (childrenTag = 'div'));
+    after(() => (childrenTag = 'span'));
```
Comment (FRSgit) [inline packages/tests/components/SfListItem/SfListItem.cy.tsx:131]: Good cleanup 👍

**Pull request merged**